### PR TITLE
Feature/rdb spec query executor

### PR DESCRIPTION
**Pull request by iceseer**
### Description of the Change
Adds specific query executor.

### CODE REVIEW

File: irohad/ametsuchi/impl/rocksdb_command_executor.cpp [hunk 1]
```diff
@@ -617,25 +646,56 @@ RocksDbCommandExecutor::ExecutionResult RocksDbCommandExecutor::operator()(
   auto const &[account_name, domain_id] = staticSplitId<2>(command.accountId());
 
   if (do_validation) {
+    uint64_t quorum;
+    if (auto result = forQuorum<kDbOperation::kGet, kDbEntry::kMustExist>(
+            common, account_name, domain_id);
+        expected::hasError(result))
+      return makeError<void>(3, std::move(result.assumeError()));
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/rocksdb_command_executor.cpp:653]: what this magic number means?

File: irohad/ametsuchi/impl/rocksdb_common.hpp [hunk 2]
```diff
@@ -712,6 +750,125 @@ namespace iroha::ametsuchi {
...
 
+  template <kDbOperation kOp,
+            typename T,
+            typename = std::enable_if_t<std::is_same<T, uint64_t>::value>>
+  inline std::optional<uint64_t> loadValue(
+      RocksDbCommon &common,
+      expected::Result<rocksdb::Status, DbError> const &status) {
+    std::optional<uint64_t> value;
+    if constexpr (kOp == kDbOperation::kGet)
+      if (status.assumeValue().ok()) {
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/rocksdb_common.hpp:761]: assumeValue() throws exception. Should it be processesed?

File: test/integration/executor/executor_fixture.hpp [hunk 3]
```diff
@@ -98,8 +98,9 @@ namespace executor_testing {
           boost::strict_get<const SpecificErrorResponse &>(&error->get()))
           << "Expected an error of type " << error_type << ", but got "
           << error->toString();
-      EXPECT_EQ(error->errorCode(), error_code)
-          << "Wrong query result error code!";
+      // TODO(iceseer): check equality of error codes for PG and RDB impls.
+      /*      EXPECT_EQ(error->errorCode(), error_code)
```
Comment (kamilsa) [inline test/integration/executor/executor_fixture.hpp:102]: do we keep this commented?

File: test/integration/executor/executor_fixture_param_rocksdb.cpp [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ */
+
+#include "integration/executor/executor_fixture_param_rocksdb.hpp"
+
+#include <boost/filesystem.hpp>
+
+#include "ametsuchi/impl/block_index_impl.hpp"
+#include "ametsuchi/impl/rocksdb_command_executor.hpp"
+#include "ametsuchi/impl/rocksdb_indexer.hpp"
+//#include "ametsuchi/impl/rocksdb_query_executor.hpp"
```
Comment (kamilsa) [inline test/integration/executor/executor_fixture_param_rocksdb.cpp:13]: remove

File: irohad/ametsuchi/impl/rocksdb_common.hpp [hunk 5]
```diff
@@ -167,6 +177,7 @@
 #define RDB_F_ASSET_SIZE "I"
 #define RDB_F_TOP_BLOCK "Q"
 #define RDB_F_PEERS_COUNT "Z"
+#define RDB_F_TOTAL_COUNT "V"
```
Comment (kuvaldini) [inline irohad/ametsuchi/impl/rocksdb_common.hpp:180]: TOTAL_TRANSACTIONS_COUNT?

File: irohad/ametsuchi/impl/rocksdb_specific_query_executor.cpp [hunk 6]
```diff
@@ -98,120 +116,732 @@ bool RocksDbSpecificQueryExecutor::hasAccountRolePermission(
...
+                  return query_response_factory_->createErrorQueryResponse(
+                      shared_model::interface::QueryResponseFactory::
+                          ErrorQueryType::kStatefulFailed,
+                      std::string("Unknown type of error happened: ")
+                          + std::to_string(error.error),
+                      1,  // unknown internal error
+                      query_hash);
+              }
+            });
+  } else {  // TODO 2019-06-06 igor-egorov IR-516 remove deprecated
```
Comment (kuvaldini) [inline irohad/ametsuchi/impl/rocksdb_specific_query_executor.cpp:787]: Is it stll actual?

File: test/integration/executor/executor_fixture.hpp [hunk 7]
```diff
@@ -98,8 +98,9 @@ namespace executor_testing {
           boost::strict_get<const SpecificErrorResponse &>(&error->get()))
           << "Expected an error of type " << error_type << ", but got "
           << error->toString();
-      EXPECT_EQ(error->errorCode(), error_code)
-          << "Wrong query result error code!";
+      // TODO(iceseer): check equality of error codes for PG and RDB impls.
```
Comment (kuvaldini) [inline test/integration/executor/executor_fixture.hpp:101]: Please create issue and link here to its number on GH

File: test/integration/executor/executor_fixture_param_rocksdb.cpp (hunk 4 above)
Comment (kuvaldini) [inline test/integration/executor/executor_fixture_param_rocksdb.cpp:13]: rm?

**Review by kuvaldini — APPROVED** [review]

File: irohad/ametsuchi/impl/rocksdb_common.hpp (hunk 5 above)
Comment (iceseer) [reply irohad/ametsuchi/impl/rocksdb_common.hpp:180]: not only transactions

**Review by kamilsa — APPROVED** [review]

**Review by kamilsa — APPROVED** [review]

File: test/integration/executor/CMakeLists.txt [hunk 8]
```diff
@@ -23,9 +23,21 @@ target_link_libraries(executor_fixture_param_postgres
     test_logger
     )
 
+add_library(executor_fixture_param_rocksdb executor_fixture_param_rocksdb.cpp)
+target_link_libraries(executor_fixture_param_rocksdb
+    ametsuchi
```
Comment (lebdron) [inline test/integration/executor/CMakeLists.txt:28]:
```suggestion
```

File: test/module/irohad/ametsuchi/wsv_query_command_test.cpp [hunk 9]
```diff
@@ -59,7 +59,7 @@ namespace iroha {
      */
     TEST_F(WsvQueryCommandTest, SetAndGetTopBlockInfo) {
       iroha::TopBlockInfo top_block_info_set{
-          1234, shared_model::crypto::Hash{"hash"}};
+          1234, shared_model::crypto::Hash{std::string("hash")}};
```
Comment (lebdron) [inline test/module/irohad/ametsuchi/wsv_query_command_test.cpp:62]: Are these changes required now that Hash has a ctor with `char const*` ?

**Review by lebdron — APPROVED** [review]
